scripts/tulipsport_sync.py

```diff
@@ -0,0 +1,212 @@
+    start_date_local = datetime.strftime(
+        summary["start_date_local"], "%Y-%m-%d %H:%M:%S"
+    )
+    # end_date = datetime.strftime(summary["end_date"], "%Y-%m-%d %H:%M:%S")
+    # end_date_local = datetime.strftime(summary["end_date_local"], "%Y-%m-%d %H:%M:%S")
+    average_heartrate = int(detail["avg_hr"])
+    map = run_map("")
+    start_latlng = None
+    distance = summary["distance"]
+    moving_time = summary["moving_time"]
+    elapsed_time = summary["elapsed_time"]
+    average_speed = summary["average_speed"]
+    location_country = ""
+
+    # 详情接口具体的内容参考文档链接：https://open.tulipsport.com/document
+    # map_data_list的结构为[ [latitude, longitude, elevation, section, distance, hr, time, cadence], ... ]
+    point_list = detail["map_data_list"]
+    point_list_length = len(point_list)
+    if point_list_length and summary["outdoor"]:
+        first_point = point_list[0]
+        start_latlng = start_point(float(first_point[0]), float(first_point[1]))
+        if point_list_length > 1:
+            last_point = point_list[-1]
+            elapsed_time = datetime.fromisoformat(
+                last_point[6]
+            ) - datetime.fromisoformat(first_point[6])
+            latlng_list = [[float(point[0]), float(point[1])] for point in point_list]
+            map = run_map(polyline.encode(latlng_list))
+
+    activity_db_instance = {
+        "id": id,
+        "name": name,
+        "type": type,
+        "start_date": start_date,
+        "start_date_local": start_date_local,
+        "average_heartrate": average_heartrate,
+        "map": map,
+        "start_latlng": start_latlng,
+        "distance": distance,
+        "moving_time": moving_time,
+        "elapsed_time": elapsed_time,
+        "average_speed": average_speed,
+        "location_country": location_country,
+    }
+    return namedtuple("activity_db_instance", activity_db_instance.keys())(
+        *activity_db_instance.values()
+    )
+
+
+def find_last_tulipsport_start_time(track_ids):
+    start_time = None
+    tulipsport_ids = [
+        id for id in track_ids if str(id).startswith(TULIPSPORT_FAKE_ID_PREFIX)
+    ]
+    if tulipsport_ids:
+        tulipsport_ids.sort()
+        # 从模拟的构造ID（特殊前缀(666) + 活动开始时间的timestamp + 活动距离(单位：米，支持单次活动最大距离为999,999米)）中读取时间信息
+        start_time = datetime.fromtimestamp(
+            int(str(tulipsport_ids[-1])[3:-6]), DEFAULT_TIMEZONE
+        )
+    return start_time
+
+
+def get_new_activities(token, old_tracks_ids):
+    s = requests.Session()
+    headers = {"Authorization": token}
+    activity_summary_list = get_all_activity_summaries(
+        s, headers, find_last_tulipsport_start_time(old_tracks_ids)
+    )
+    activity_summary_list = [
+        activity
+        for activity in activity_summary_list
+        if activity["id"] not in old_tracks_ids
+    ]
+    print(f"{len(activity_summary_list)} new activities to generate")
+    tracks = []
+    old_gpx_ids = os.listdir(GPX_FOLDER)
```

Comment:
To stay true to the original design, consider add the `--with-gpx` parameter. You can search for this parameter within the codebase to learn more.

Reply:
OK, I will find out later~

Reply:
@ben-29 It's done~
https://github.com/yihong0618/running_page/pull/379/commits/be3a9af16e9297269860f03c40b4366c8f9b98a5